### PR TITLE
Update distro matrix for CI

### PR DESCRIPTION
**Pull request by omajid**

### CODE REVIEW

**Comment by omajid** [comment]
Hey @ayakael I am trying to get these tests to pass on Alpine and seeing some unexpected failures.

### `liblttng-ust_sys-sdt.h`

```
/ # readelf -n /usr/lib/liblttng-ust.so.1

Displaying notes found in: .note.gnu.property
  Owner                Data size        Description
  GNU                  0x00000020       NT_GNU_PROPERTY_TYPE_0
      Properties: x86 feature used: x86, XMM
        x86 ISA used: x86-64-baseline
```

No `NT_STAPSDT` notes here? Should we just exclude this test on alpine?

### `workload`

As a non-root user, `dotnet workload install` fails:

```
$  id
uid=1000(omajid) gid=1000(omajid) groups=1000(omajid)
$ dotnet workload install wasm-tools
Inadequate permissions. Run the command with elevated privilege
```

This works fine on Fedora/RHEL out of the box. Is this failure known/intentional? Should we skip this test on alpine?

**Comment by ayakael** [comment]
> Hey @ayakael I am trying to get these tests to pass on Alpine and seeing some unexpected failures.
> 
> ### `liblttng-ust_sys-sdt.h`
> ```
> / # readelf -n /usr/lib/liblttng-ust.so.1
> 
> Displaying notes found in: .note.gnu.property
>   Owner                Data size        Description
>   GNU                  0x00000020       NT_GNU_PROPERTY_TYPE_0
>       Properties: x86 feature used: x86, XMM
>         x86 ISA used: x86-64-baseline
> ```
> Hey @ayakael I am trying to get these tests to pass on Alpine and seeing some unexpected failures.

### `liblttng-ust_sys-sdt.h`
```
/ # readelf -n /usr/lib/liblttng-ust.so.1

Displaying notes found in: .note.gnu.property
  Owner                Data size        Description
  GNU                  0x00000020       NT_GNU_PROPERTY_TYPE_0
      Properties: x86 feature used: x86, XMM
        x86 ISA used: x86-64-baseline
```

No `NT_STAPSDT` notes here? Should we just exclude this test on alpine?

### `workload`
As a non-root user, `dotnet workload install` fails:

```
$  id
uid=1000(omajid) gid=1000(omajid) groups=1000(omajid)
$ dotnet workload install wasm-tools
Inadequate permissions. Run the command with elevated privilege
```

This works fine on Fedora/RHEL out of the box. Is this failure known/intentional? Should we skip this test on alpine?
> No `NT_STAPSDT` notes here? Should we just exclude this test on alpine?
> 
> ### `workload`
> As a non-root user, `dotnet workload install` fails:
> 
> ```
> $  id
> uid=1000(omajid) gid=1000(omajid) groups=1000(omajid)
> $ dotnet workload install wasm-tools
> Inadequate permissions. Run the command with elevated privilege
> ```
> 
> This works fine on Fedora/RHEL out of the box. Is this failure known/intentional? Should we skip this test on alpine?

Indeed, these two test units are disabled on our side. `liblttng-ust` as it is broken anyways due to https://github.com/dotnet/runtime/issues/57784, and `workload` due to the aforementionned permission issue that I've yet to find time to investigate.

**Comment by ayakael** [comment]
For reference, here are other test units we disable on our side:

* man-pages distribution-package bash-completion install-local 
Disabled as the test suite occurs before packaging step
* release-version-sane
We don't really care of the release version is sane. It causes issues previously so we just disable it.
* omnisharp
No 7.0 release yet
* createdump-aspnet workload
Permission issue on lxc / pipelines
* {bundled,system}-libunwind
Disabled / enabled depending on platform as `aarch64` / `armv7` is problematic and mono-flavored runtime doesn't link against libunwind.

**Comment by omajid** [comment]
> For reference, here are other test units we disable on our side:
> ...

If you want, we can mark these as tests to skip on alpine (on particular archs) here in this repo. We do that for a number of os/arch combinations already.

File: runtime-id [hunk 1]
```diff
@@ -47,9 +47,22 @@ if [[ ${portable_rid} == 1 ]]; then
...
+        ol|rhel|rocky)
+            rid_version=${VERSION_ID%.*}
+            ;;
+        alpine)
+            if [[ $VERSION_ID == *"_"* ]]; then
+                # Alpine in-development versions contain underscore and no third part of the version
+                rid_version=${VERSION_ID%_*}
+            else
+                # Remove the patch version from major.minor.micro string
+                rid_version=${VERSION_ID%.*}
```
Comment (ayakael) [inline runtime-id:59]: Thanks for this. This is not always true anymore. Sometimes it is '3.17.0`, sometimes `3.17.0_rc1`, sometimes `3.17_alpha`. To cover all possibilities, we do the following to normalize all possible RIDs:
```diff
diff --git a/runtime-id b/runtime-id
index 5673bba..7369776 100755
--- a/runtime-id
+++ b/runtime-id
@@ -44,12 +44,15 @@ if [[ ${portable_rid} == 1 ]]; then
         echo "linux-${arch}"
     fi
 else
-    case "${ID}" in
-        # Remove the minor version
-        alpine|ol|rhel|rocky) rid_version=${VERSION_ID%.*} ;;
-
-        *) rid_version=${VERSION_ID} ;;
-    esac
-
-    echo "${ID}.${rid_version}-${arch}"
+    VERSION_ID_DOT="${VERSION_ID//[^.]}"
+    while [[ ${#VERSION_ID_DOT} -gt 1 ]]; do
+        VERSION_ID="${VERSION_ID%.*}"
+        VERSION_ID_DOT="${VERSION_ID//[^.]}"
+    done
+    VERSION_ID_DASH="${VERSION_ID//[^_]}"
+    while [[ ${#VERSION_ID_DASH} -ge 1 ]]; do
+        VERSION_ID="${VERSION_ID%_*}"
+        VERSION_ID_DASH="${VERSION_ID//[^_]}"
+    done
+    echo "${ID}.${VERSION_ID}-${arch}"
 fi
-- 
2.37.1
```
It'd be nice if RID normalization logics were universalized as, indeed, hardcoding DistroWatch in our code is becoming a chore.

Comment (omajid) [reply runtime-id:59]: > hardcoding DistroWatch in our code is becoming a chore

:rofl:

**Comment by ayakael** [comment]
> > For reference, here are other test units we disable on our side:
> > ...
> 
> If you want, we can mark these as tests to skip on alpine (on particular archs) here in this repo. We do that for a number of os/arch combinations already.
> 
> Edit: Maybe not for the pre-vs-post-package tests; the tests here are really meant to be execute post-packaging. But things like bundled/system-libunwind can be added here no problem.
> 
> Also, omnisharp is already disabled on .NET 7

For libunwind:
bundled-libunwind for: x86_64, x86
system-libunwind for: armv7, aarch64 due to being broken
{bundled,system}-libunwind for all mono-flavored platforms: s390x, ppc64le, armhf

Thanks for omnisharp, I'll remove those logics :)

**Comment by omajid** [comment]
The recently-added test to verify that PostgreSQL works with .NET is failing on Alpine.

For edge, we get an error saying the platform is just not supported.

```
Unhandled exception. System.PlatformNotSupportedException: System.Data.ODBC is not supported on this platform.
   at System.Data.Odbc.OdbcConnection..ctor(String connectionString)
   at Program.<Main>$(String[] args) in /__w/dotnet-regular-tests/dotnet-regular-tests/system-data-
```

For latest, we get a stack trace:

```
 Unhandled exception. System.Data.Odbc.OdbcException (0x80131937): ERROR [01000] [unixODBC][Driver Manager]Can't open lib 'PostgreSQL' : file not found
   at System.Data.Odbc.OdbcConnection.HandleError(OdbcHandle hrHandle, RetCode retcode)
   at System.Data.Odbc.OdbcConnectionHandle..ctor(OdbcConnection connection, OdbcConnectionString constr, OdbcEnvironmentHandle environmentHandle)
   at System.Data.Odbc.OdbcConnectionOpen..ctor(OdbcConnection outerConnection, OdbcConnectionString connectionOptions)
   at System.Data.Odbc.OdbcConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningObject)
   at System.Data.ProviderBase.DbConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreateNonPooledConnection(DbConnection owningConnection, DbConnectionPoolGroup poolGroup, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionInternal.OpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory)
   at System.Data.Odbc.OdbcConnection.Open()
   at Program.<Main>$(String[] args) in /__w/dotnet-regular-tests/dotnet-regular-tests/system-data-odbc/Program.cs:line 14
```

Am I missing a library? Are these known issues? Should I skip these tests on alpine?

**Comment by ayakael** [comment]
> The recently-added test to verify [that System.Data.Odbc/PostgreSQL works with .NET](https://github.com/redhat-developer/dotnet-regular-tests/blob/main/system-data-odbc/test.sh) is failing on Alpine.
> 
> For edge, we get an error saying the platform is just not supported.
> 
> ```
> Unhandled exception. System.PlatformNotSupportedException: System.Data.ODBC is not supported on this platform.
>    at System.Data.Odbc.OdbcConnection..ctor(String connectionString)
>    at Program.<Main>$(String[] args) in /__w/dotnet-regular-tests/dotnet-regular-tests/system-data-
> ```
> 
> For latest, we get a stack trace:
> 
> ```
>  Unhandled exception. System.Data.Odbc.OdbcException (0x80131937): ERROR [01000] [unixODBC][Driver Manager]Can't open lib 'PostgreSQL' : file not found
>    at System.Data.Odbc.OdbcConnection.HandleError(OdbcHandle hrHandle, RetCode retcode)
>    at System.Data.Odbc.OdbcConnectionHandle..ctor(OdbcConnection connection, OdbcConnectionString constr, OdbcEnvironmentHandle environmentHandle)
>    at System.Data.Odbc.OdbcConnectionOpen..ctor(OdbcConnection outerConnection, OdbcConnectionString connectionOptions)
>    at System.Data.Odbc.OdbcConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningObject)
>    at System.Data.ProviderBase.DbConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
>    at System.Data.ProviderBase.DbConnectionFactory.CreateNonPooledConnection(DbConnection owningConnection, DbConnectionPoolGroup poolGroup, DbConnectionOptions userOptions)
>    at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
>    at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
>    at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
>    at System.Data.ProviderBase.DbConnectionInternal.OpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory)
>    at System.Data.Odbc.OdbcConnection.Open()
>    at Program.<Main>$(String[] args) in /__w/dotnet-regular-tests/dotnet-regular-tests/system-data-odbc/Program.cs:line 14
> ```
> 
> Am I missing a library? Are these known issues? Should I skip these tests on alpine?

Reproduced on my side, investigating further...

**Comment by ayakael** [comment]
Writing the following to `/etc/odbcinst.ini` and installing the `psqlodbc` package fixes the error:
```
[PostgreSQL Unicode]
Description=PostgreSQL Unicode
Driver=/usr/lib/psqlodbcw.so
UsageCount=1
```

**Comment by ayakael** [comment]
For `system-data-odbc` on `net6` failing, would you have an idea on why the platform wouldn't be supported? Otherwise, I'm tempted to disable for `net6` on Alpine, as `net7` works.

**Comment by omajid** [comment]
It's the version string in `/etc/os-release`:

```
# cat /etc/os-release 
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.17_alpha20221110
PRETTY_NAME="Alpine Linux edge"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
# PGHOST=localhost PGPORT=8080 EXPECTED_ROWS=2 EXPECTED_COLUMNS=2  dotnet run
DRIVER={PostgreSQL};SERVER=localhost; PORT=8080;DATABASE=testdb;
Unhandled exception. System.PlatformNotSupportedException: System.Data.ODBC is not supported on this platform.
   at System.Data.Odbc.OdbcConnection..ctor(String connectionString)
   at Program.<Main>$(String[] args) in /root/dotnet-regular-tests/system-data-odbc/Program.cs:line 13
# vi /etc/os-release 
# cat /etc/os-release 
NAME="Alpine Linux"
ID=alpine
#VERSION_ID=3.17_alpha20221110
VERSION_ID=3.17
PRETTY_NAME="Alpine Linux edge"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
~/dotnet-regular-tests/system-data-odbc # PGHOST=localhost PGPORT=8080 EXPECTED_ROWS=2 EXPECTED_COLUMNS=2  dotnet run
DRIVER={PostgreSQL};SERVER=localhost; PORT=8080;DATABASE=testdb;
Unhandled exception. System.DllNotFoundException: Dependency unixODBC with minimum version 2.3.1 is required.
Unable to load shared library 'libodbc.so.2' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: Error loading shared library liblibodbc.so.2: No such file or directory
   at System.Data.Odbc.OdbcConnection.Open()
   at Program.<Main>$(String[] args) in /root/dotnet-regular-tests/system-data-odbc/Program.cs:line 14
```

**Comment by ayakael** [comment]
> It's the version string in `/etc/os-release`:
> 
> ```
> # cat /etc/os-release 
> NAME="Alpine Linux"
> ID=alpine
> VERSION_ID=3.17_alpha20221110
> PRETTY_NAME="Alpine Linux edge"
> HOME_URL="https://alpinelinux.org/"
> BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
> # PGHOST=localhost PGPORT=8080 EXPECTED_ROWS=2 EXPECTED_COLUMNS=2  dotnet run
> DRIVER={PostgreSQL};SERVER=localhost; PORT=8080;DATABASE=testdb;
> Unhandled exception. System.PlatformNotSupportedException: System.Data.ODBC is not supported on this platform.
>    at System.Data.Odbc.OdbcConnection..ctor(String connectionString)
>    at Program.<Main>$(String[] args) in /root/dotnet-regular-tests/system-data-odbc/Program.cs:line 13
> # vi /etc/os-release 
> # cat /etc/os-release 
> NAME="Alpine Linux"
> ID=alpine
> #VERSION_ID=3.17_alpha20221110
> VERSION_ID=3.17
> PRETTY_NAME="Alpine Linux edge"
> HOME_URL="https://alpinelinux.org/"
> BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
> ~/dotnet-regular-tests/system-data-odbc # PGHOST=localhost PGPORT=8080 EXPECTED_ROWS=2 EXPECTED_COLUMNS=2  dotnet run
> DRIVER={PostgreSQL};SERVER=localhost; PORT=8080;DATABASE=testdb;
> Unhandled exception. System.DllNotFoundException: Dependency unixODBC with minimum version 2.3.1 is required.
> Unable to load shared library 'libodbc.so.2' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: Error loading shared library liblibodbc.so.2: No such file or directory
>    at System.Data.Odbc.OdbcConnection.Open()
>    at Program.<Main>$(String[] args) in /root/dotnet-regular-tests/system-data-odbc/Program.cs:line 14
> ```

Christ on a broomstick, I wish I knew how to fix RID parsing... it's being tracked in https://github.com/dotnet/runtime/issues/73525.

**Comment by omajid** [comment]
One thing that's still unclear to me is the relationship between the RID in `dotnet --info` and what's in `/etc/os-release`. I (naively) expected that dotnet would use the RID it identifies to do dll resolution?

```
# dotnet --info                                                                                                                                             
.NET SDK (reflecting any global.json):                                                                                                                        
 Version:   6.0.113                                                                                                                                           
 Commit:    4a23b50f97                                                                                                                                        
                                                                                                                                                              
Runtime Environment:                                                                                                                                          
 OS Name:     alpine                                                                                                                                          
 OS Version:  3.17_alpha20221110                                                                                                                              
 OS Platform: Linux                                                                                                                                           
 RID:         alpine.3.17-x64                                                                                                                                 
 Base Path:   /usr/lib/dotnet/sdk/6.0.113/                                                                                                                    
                                                                                                                                                              
global.json file:                                                                                                                                             
  Not found                                                                                                                                                   
                                                                                                                                                              
Host:                                                                                                                                                         
  Version:      6.0.13                                                                                                                                        
  Architecture: x64                                                                                                                                           
  Commit:       1af80ba017                                                                                                                                    
                                                                                                                                                              
.NET SDKs installed:                                                                                                                                          
  6.0.113 [/usr/lib/dotnet/sdk]                                                                                                                               
                                                                                                                                                              
.NET runtimes installed:                                                                                                                                      
  Microsoft.AspNetCore.App 6.0.13 [/usr/lib/dotnet/shared/Microsoft.AspNetCore.App]                                                                           
  Microsoft.NETCore.App 6.0.13 [/usr/lib/dotnet/shared/Microsoft.NETCore.App]                                                                                 
                                                                                                                                                              
Download .NET:                                                                                                                                                
  https://aka.ms/dotnet-download                                                                                                                              
                                                                                                                                                              
Learn about .NET Runtimes and SDKs:                                                                                                                           
  https://aka.ms/dotnet/runtimes-sdk-info
```

**Pull request merged**